### ISSUE TITLE
MISP upgrade to the version v2.4.149 finished with errors regarding misp-stix

### ISSUE THREAD

**Issue by samitainio** [MISP/MISP#7812]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu 20.04.3 LTS
| PHP version               | 7.4.24
| MISP version / git hash   | v2.4.149 (85f8912f8d159b141f8f74514af374eb1487a643)

### Expected behavior
MISP version upgrade would finish without errors.

### Actual behavior
Errors, see the log below.

### Steps to reproduce the behavior
Go to the diagnostics page and update MISP.

### Logs, screenshots, configuration dump, ...

Here is a snippet of the diagnostics page UI log immediately after the update:

```
cd $(git rev-parse --show-toplevel) && git submodule update --init --recursive 2>&1

Submodule 'app/files/scripts/misp-stix' (git@github.com:misp/misp-stix.git) registered for path 'app/files/scripts/misp-stix'
Submodule 'app/files/scripts/mixbox' (https://github.com/CybOXProject/mixbox) registered for path 'app/files/scripts/mixbox'
Submodule 'app/files/scripts/python-cybox' (https://github.com/CybOXProject/python-cybox) registered for path 'app/files/scripts/python-cybox'
Submodule 'app/files/scripts/python-maec' (https://github.com/MAECProject/python-maec) registered for path 'app/files/scripts/python-maec'
Submodule 'app/files/scripts/python-stix' (https://github.com/STIXProject/python-stix) registered for path 'app/files/scripts/python-stix'
Cloning into '/var/www/MISP/app/files/scripts/misp-stix'...
Could not create directory '/var/www/.ssh'.
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:misp/misp-stix.git' into submodule path '/var/www/MISP/app/files/scripts/misp-stix' failed
Failed to clone 'app/files/scripts/misp-stix'. Retry scheduled
Cloning into '/var/www/MISP/app/files/scripts/misp-stix'...
Could not create directory '/var/www/.ssh'.
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:misp/misp-stix.git' into submodule path '/var/www/MISP/app/files/scripts/misp-stix' failed
Failed to clone 'app/files/scripts/misp-stix' a second time, aborting
```

**Comment by torsolaso** [comment]
same behaviour here. same logs

**Comment by torsolaso** [comment]
```
2021-10-08T07:36:40.196969728Z Cloning into '/var/www/MISP/PyMISP/pymisp/data/misp-objects'...
2021-10-08T07:36:41.298418173Z Cloning into '/var/www/MISP/PyMISP/pymisp/tools/pdf_fonts'...
2021-10-08T07:36:48.684991734Z Submodule path 'PyMISP/pymisp/data/misp-objects': checked out '238fc99b6019db9b41185794fe53590af69a17d1'
2021-10-08T07:36:49.976938175Z Submodule path 'PyMISP/pymisp/tools/pdf_fonts': checked out '7ff222022e6ad99e11a201f62d57da4bff1337ee'
2021-10-08T07:36:50.221663301Z Submodule path 'app/files/misp-galaxy': checked out '20f14c0090f1963ee485b98fa99c62f7830f734f'
2021-10-08T07:36:50.328376397Z Submodule path 'app/files/misp-objects': checked out '1827b14c20e31355b1216fa896e6f8dad62ec12b'
2021-10-08T07:36:50.485250668Z Submodule path 'app/files/scripts/cti-python-stix2': checked out '519a5cb8fad2fdfd715ca32bcd5b7f414e395a59'
2021-10-08T07:36:50.561405036Z Submodule path 'app/files/scripts/misp-opendata': checked out 'f06333e3e834e37a8efe3d24105b51797ccc5a6c'
2021-10-08T07:36:50.643741951Z Submodule path 'app/files/scripts/misp-stix': checked out '8cfdcb04db3418636fa1c39f7f8655e91b160b50'
2021-10-08T07:36:50.649449693Z Submodule 'cti' (git@github.com:mitre/cti.git) registered for path 'app/files/scripts/misp-stix/cti'
2021-10-08T07:36:50.657884856Z Cloning into '/var/www/MISP/app/files/scripts/misp-stix/cti'...
2021-10-08T07:36:50.954246368Z Host key verification failed.
2021-10-08T07:36:50.954302168Z fatal: Could not read from remote repository.
2021-10-08T07:36:50.954309968Z 
2021-10-08T07:36:50.954330368Z Please make sure you have the correct access rights
2021-10-08T07:36:50.954334668Z and the repository exists.
2021-10-08T07:36:50.958255798Z fatal: clone of 'git@github.com:mitre/cti.git' into submodule path '/var/www/MISP/app/files/scripts/misp-stix/cti' failed
2021-10-08T07:36:50.959002603Z Failed to clone 'cti'. Retry scheduled
2021-10-08T07:36:50.967165964Z Cloning into '/var/www/MISP/app/files/scripts/misp-stix/cti'...
2021-10-08T07:36:51.265324289Z Host key verification failed.
2021-10-08T07:36:51.266322997Z fatal: Could not read from remote repository.
2021-10-08T07:36:51.266359697Z 
2021-10-08T07:36:51.266382097Z Please make sure you have the correct access rights
2021-10-08T07:36:51.266386397Z and the repository exists.
2021-10-08T07:36:51.269221518Z fatal: clone of 'git@github.com:mitre/cti.git' into submodule path '/var/www/MISP/app/files/scripts/misp-stix/cti' failed
2021-10-08T07:36:51.269239918Z Failed to clone 'cti' a second time, aborting
2021-10-08T07:36:51.328516561Z Submodule path 'app/files/scripts/mixbox': checked out '70bc18c714134c5be5a70226aac5961ad810a6c0'
2021-10-08T07:36:51.477005369Z Submodule path 'app/files/scripts/python-cybox': checked out '25e6e8b3a6f429f079d3fbd9ace3db9eb3d5ab71'
2021-10-08T07:36:51.570685268Z Submodule path 'app/files/scripts/python-maec': checked out '13e66105c9646156060d0896a4d54970ea358f44'
2021-10-08T07:36:51.670262711Z Submodule path 'app/files/scripts/python-stix': checked out '34f24261b70c7d02ddf0f5696d3919e771dbd32f'
2021-10-08T07:36:51.766132726Z Submodule path 'app/files/taxonomies': checked out '9198e5f022dcad03ecdae7cc72689176a1590f52'
2021-10-08T07:36:52.480152355Z Submodule path 'app/files/warninglists': checked out '27fb7a6a44a06039d7a5d8b35c405db79bf4384a'
2021-10-08T07:36:52.532938949Z Submodule path 'misp-vagrant': checked out '8622be9f3074ead2e315803aac94eec6f16461b5'
2021-10-08T07:36:52.544126932Z Failed to recurse into submodule path 'app/files/scripts/misp-stix'
```

**Comment by m4rb3n** [comment]
Same Issue

**Comment by adulau** [comment]
Git pull to the latest version. It should be fixed. Thanks for the report.

**Comment by hadisultan** [comment]
Getting the same issue despite git pull to the latest version. I've tried installing on ubuntu 18, ubuntu 20, kali and docker.

EDIT: In my case, I'm trying to install MISP from scratch, not just update it. So maybe the install script needs fixing as well.

**Comment by feryandi** [comment]
I am also having the exact same issue when trying to install MISP. Here's some error snippet:
```
Cloning into '/var/www/MISP/PyMISP/pymisp/tools/pdf_fonts'...
remote: Enumerating objects: 16, done.        
remote: Total 16 (delta 0), reused 0 (delta 0), pack-reused 16        
Submodule path 'PyMISP/pymisp/data/misp-objects': checked out '3d52773e9d3ba39ff324455bf8c10b47e11b695a'
Submodule path 'PyMISP/pymisp/tools/pdf_fonts': checked out '7ff222022e6ad99e11a201f62d57da4bff1337ee'
Submodule path 'app/Lib/cakephp': checked out 'cf14e6546ec44e3369e3531add11fdb946656280'
Submodule path 'app/Lib/random_compat': checked out '088c04e2f261c33bed6ca5245491cfca69195ccf'
Submodule path 'app/files/misp-decaying-models': checked out 'f1b974790f0969fab4251046f292eedc59a272ca'
Submodule path 'app/files/misp-galaxy': checked out '20f14c0090f1963ee485b98fa99c62f7830f734f'
Submodule path 'app/files/misp-objects': checked out '1827b14c20e31355b1216fa896e6f8dad62ec12b'
Submodule path 'app/files/noticelists': checked out '028f569e26b5b47286167b7d39f81cd1791eb04e'
Submodule path 'app/files/scripts/cti-python-stix2': checked out '519a5cb8fad2fdfd715ca32bcd5b7f414e395a59'
Submodule path 'app/files/scripts/misp-opendata': checked out 'f06333e3e834e37a8efe3d24105b51797ccc5a6c'
Submodule path 'app/files/scripts/misp-stix': checked out '83d9300edeb94ac98fad6f8feb3e2640f583bb59'
Submodule 'cti' (git@github.com:mitre/cti.git) registered for path 'app/files/scripts/misp-stix/cti'
Cloning into '/var/www/MISP/app/files/scripts/misp-stix/cti'...
Could not create directory '/var/www/.ssh'.
The authenticity of host 'github.com (140.82.114.4)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? YES
Failed to add the host to the list of known hosts (/var/www/.ssh/known_hosts).
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```

**Comment by feryandi** [comment]
I also have the same issue when trying to install MISP in Ubuntu 18, here's some snippet of the error:
```
#############################  (28%)
Next step: Installing MISP core
Next step: Trying to git pull existing install
fatal: not a git repository (or any of the parent directories): .git
Cloning into '/var/www/MISP/app/files/scripts/misp-stix/cti'...
Could not create directory '/var/www/.ssh'.
The authenticity of host 'github.com (140.82.113.3)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? yes
Failed to add the host to the list of known hosts (/var/www/.ssh/known_hosts).
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
 
Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:mitre/cti.git' into submodule path '/var/www/MISP/app/files/scripts/misp-stix/cti' failed
Failed to clone 'cti'. Retry scheduled
Cloning into '/var/www/MISP/app/files/scripts/misp-stix/cti'...
Could not create directory '/var/www/.ssh'.
The authenticity of host 'github.com (140.82.113.3)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? yes
Failed to add the host to the list of known hosts (/var/www/.ssh/known_hosts).
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```

**Comment by adulau** [comment]
In `MISP/app/files/scripts`, don’t forget to do a `git submodule init` and `git submodule update`.

Fixed in 932a3ec28b76b7e7e856a8f418c0137066bc8fc2

**Comment by samitainio** [comment]
@adulau 

I updated the MISP to the latest commit. However some additional steps were required. I had to also manually edit the `/var/www/MISP/.git/config` file since it still had the incorrect value in place:

```
[submodule "app/files/scripts/misp-stix"]
        active = true
        url = git@github.com:misp/misp-stix.git
```

I changed it into

```
[submodule "app/files/scripts/misp-stix"]
        active = true
        url = https://github.com/misp/misp-stix.git
```

After the edit I ran the commands:

```
cd /var/www/MISP
sudo -u www-data git submodule update --init --recursive
```

It now completed succesfully.
